### PR TITLE
Fix compilation of jsk_rviz_plugins for ROS noetic

### PR DESCRIPTION
**Pull request by Rayman**
I've added ifdefs for older ROS versions so there is no change there.

If we want, we could switch to tf2 at an earlier distro, for example already in ROS melodic.

### CODE REVIEW

File: jsk_rviz_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,14 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(jsk_rviz_plugins)
 if("$ENV{ROS_DISTRO}" STRGREATER "indigo")
-  add_compile_options(-std=c++11)
+  if("$ENV{ROS_DISTRO}" STRGREATER "melodic")
+    # How to require c++14: https://stackoverflow.com/a/42842447
+    set(CMAKE_CXX_STANDARD 14)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_EXTENSIONS OFF)
```
Comment (knorth55) [inline jsk_rviz_plugins/CMakeLists.txt:12]: `CMAKE_CXX_STANDARD` is ntroduced in `cmake>=3.1`
please use `add_compile_options(-std=c++14)`

**Comment by knorth55** [comment]
Thank you for PR.
we will work on noetic support, but it will take long time because even if we can build this package with noetic, we have to solve python2/3 problems in ROS noetic.
So, noetic support requires some time and we are waiting for other dependencies supporting noetic.
For full noetic support, please wait but also we are welcoming for new PRs.

**Comment by Rayman** [comment]
Yes, full noetic support might take a while, but each patch helps a little I hope :)

**Comment by knorth55** [comment]
> Yes, full noetic support might take a while, but each patch helps a little I hope :)

Ofcourese! PRs help us a lot.
But we need to set up noetic test build first, so it will take a while to merge this PR.

**Comment by PeterMitrano** [comment]
I ran into this same issue (this PR seems to fix it). But just as some unsolicited CMake advice, I would use `set(CMAKE_CXX_STANDARD 17)` or 14 or whatever instead of setting compiler flags directly. This way CMake can set flags for different OS's and compilers correctly.

**Comment by Rayman** [comment]
@PeterMitrano, that's what I did first, but @knorth55 mentioned that this feature is not available for cmake<3.1 so I changed this.

I'm glad it works, hopefully it will get merged soon.

**Comment by PeterMitrano** [comment]
ah what a pity. I didn't imagine anyone could still be using cmake <3.1 but ok!

**Comment by knorth55** [comment]
@PeterMitrano we use this package on ROS indigo because our robot is still running on ROS Indigo.
we want to upgrade it to ROS Melodic or Noetic, but we don't have enough time to do so.

**Pull request merged**